packages/scandipwa/src/component/CartCoupon/CartCoupon.component.js

```diff
@@ -47,11 +49,26 @@ export class CartCoupon extends PureComponent {
 
     handleFormSubmit = this.handleFormSubmit.bind(this);
 
+    getSnapshotBeforeUpdate(prevProps) {
+        const { isBadCoupon: prevIsBadCoupon = false } = prevProps;
+        const { isBadCoupon = false, resetIsBadCoupon } = this.props;
+
+        if (isBadCoupon && prevIsBadCoupon !== isBadCoupon) {
+            this.setState({
+                isFieldWithError: isBadCoupon
+            });
+            resetIsBadCoupon();
+        }
+    }
+
+    componentDidUpdate() {}
```

Comment:
Why didn't you put this logic to `componentDidUpdate`?

Reply:
This isn't correct use case for `getSnapshotBeforeUpdate`.

Reply:
getSnapshotBeforeUpdate was a workaround for the error "you can't use setState in componentDidUpdate"

But I changed it - factored out setState in another function